src/main/java/frc/team2412/robot/Subsystems/IndexerSubsystem.java

```diff
@@ -0,0 +1,304 @@
+package frc.team2412.robot.Subsystems;
+
+import com.revrobotics.CANSparkMax;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.team2412.robot.RobotMap;
+
+public class IndexerSubsystem extends SubsystemBase {
+	public CANSparkMax indexBackMotor;
+	public CANSparkMax indexFrontMotor;
+	public CANSparkMax indexMidMotor;
+	public SpeedControllerGroup indexMotors;
+	public SpeedControllerGroup indexSideMotors;
+
+	public DigitalInput back;
+	public DigitalInput backMid;
+	public DigitalInput mid;
+	public DigitalInput frontMid;
+	public DigitalInput front;
+	
+	public DoubleSolenoid frontClutch;
+	public DoubleSolenoid rearClutch;
+
+	public int numBalls = 0;
+	public boolean ballUnbalancedSide = false;
+
+	public IndexerSubsystem() {
+		// Motors
+		indexBackMotor = RobotMap.indexFrontMotor;
+		indexFrontMotor = RobotMap.indexBackMotor;
+		indexMidMotor = RobotMap.indexMidMotor;
+		indexMotors = new SpeedControllerGroup(indexFrontMotor, indexBackMotor, indexMidMotor);
+		indexSideMotors = new SpeedControllerGroup(indexFrontMotor, indexBackMotor);
+
+		// Sensors
+		back = RobotMap.BB;
+		backMid = RobotMap.BT;
+		mid = RobotMap.T;
+		frontMid = RobotMap.FT;
+		front = RobotMap.FF;
+		
+		//soulenoid
+		frontClutch = RobotMap.frontClutch;
+		rearClutch = RobotMap.rearClutch;
+	}
+
+	public void indexerLogic(boolean shoot) {
+		if (shoot) {
+			shoot();

```

Comment:
This should be a subsystem method all by itself, and not embedded in something else.